In [9]:
import pandas as pd

In [10]:
TFL = pd.read_csv('data/tfl_journeys_cleaned.csv')

In [11]:
%%sql
SELECT 
    JOURNEY_TYPE,
    SUM(JOURNEYS_MILLIONS) AS Total_Journeys_Millions
FROM 
    TFL
GROUP BY 
    JOURNEY_TYPE
ORDER BY 
    Total_Journeys_Millions DESC;

journey_type  Total_Journeys_Millions
0  Underground & DLR             15020.466547
1                Bus              7015.592766
2         Overground              1666.845665
3           TfL Rail               411.313420
4               Tram               314.689871
5   Emirates Airline                14.583717

In [12]:
%%sql
SELECT 
    CASE 
        WHEN DAYOFWEEK(CAST(REPORT_DATE AS DATE)) IN (1, 7) THEN 'Weekend'
        ELSE 'Weekday'
    END AS Day_Type,
    AVG(JOURNEYS_MILLIONS) AS Average_Journeys_Millions
FROM 
    TFL
GROUP BY 
    Day_Type;

Day_Type  Average_Journeys_Millions
0  Weekday                  32.755662
1  Weekend                  34.427841

### Comparative Analysis of Transport Modes

#### Overview of Journey Volumes

- **Underground & DLR**: The most heavily utilized transport mode, with approximately 15,020.47 million journeys over the analyzed period. The London Underground and Docklands Light Railway are essential components of London's public transport network, catering to a significant portion of commuters and travelers.

- **Bus**: The second most utilized mode of transport, with approximately 7,015.59 million journeys. Buses play a crucial role in providing accessibility to areas not served by other modes of transport and complementing the Underground and Overground services.

- **Overground**: Follows with approximately 1,666.85 million journeys, providing vital connectivity across various parts of London, particularly in areas not well-served by the Underground network.

- **TfL Rail**, **Tram**, and **Emirates Airline**: Each contribute to the overall transport network, serving specific niches and providing connectivity to different parts of the city.

### Conclusion

The analysis reveals the diverse range of transport modes available in London, each fulfilling specific travel needs and serving different parts of the city. The Underground & DLR and bus services emerge as the most popular modes, reflecting their widespread coverage and accessibility.

The data underscores the importance of a comprehensive public transport network in facilitating mobility and connectivity within urban environments. London's transport infrastructure plays a vital role in supporting economic growth, reducing traffic congestion, and enhancing the quality of life for residents and visitors alike.